src/unit_tests/wazuh_db/test_wdb_agents_helpers.c

```diff
@@ -339,6 +339,180 @@ void test_wdb_agents_vuln_cve_clear_success(void **state)
+    int ret = 0;
+    int id = 1;
+    const char *old_status = "valid";
+    const char *new_status = "obsolete";
+
+    const char *json_str = strdup("{\"old_status\":\"valid\",\"new_status\":\"obsolete\"}");
+    const char *query_str = "agent 1 vuln_cve update_status {\"old_status\":\"valid\",\"new_status\":\"obsolete\"}";
+    const char *response = "err";
+
+    will_return(__wrap_cJSON_CreateObject, 1);
+    will_return_always(__wrap_cJSON_AddStringToObject, 1);
+
+    // Adding data to JSON
+    expect_string(__wrap_cJSON_AddStringToObject, name, "old_status");
+    expect_string(__wrap_cJSON_AddStringToObject, string, "valid");
+    expect_string(__wrap_cJSON_AddStringToObject, name, "new_status");
+    expect_string(__wrap_cJSON_AddStringToObject, string, "obsolete");
+
+    // Printing JSON
+    will_return(__wrap_cJSON_PrintUnformatted, json_str);
+    expect_function_call(__wrap_cJSON_Delete);
+
+    // Calling Wazuh DB
+    expect_any(__wrap_wdbc_query_ex, *sock);
+    expect_string(__wrap_wdbc_query_ex, query, query_str);
+    expect_value(__wrap_wdbc_query_ex, len, WDBOUTPUT_SIZE);
+    will_return(__wrap_wdbc_query_ex, response);
+    will_return(__wrap_wdbc_query_ex, OS_INVALID);
+
+    // Handling result
+    expect_string(__wrap__mdebug1, formatted_msg, "Agents DB (1) Error in the response from socket");
+    expect_string(__wrap__mdebug2, formatted_msg, "Agents DB (1) SQL query: agent 1 vuln_cve update_status {\"old_status\":\"valid\",\"new_status\":\"obsolete\"}");
+
+    ret = wdb_agents_vuln_cve_update_status(id, old_status, new_status, NULL);
+
+    assert_int_equal(OS_INVALID, ret);
+}
+
+test_wdb_agents_vuln_cve_update_status_error_sql_execution(void **state){
+    int ret = 0;
+    int id = 1;
+    const char *old_status = "valid";
+    const char *new_status = "obsolete";
+
+    const char *json_str = strdup("{\"old_status\":\"valid\",\"new_status\":\"obsolete\"}");
+    const char *query_str = "agent 1 vuln_cve update_status {\"old_status\":\"valid\",\"new_status\":\"obsolete\"}";
+    const char *response = "err";
+
+    will_return(__wrap_cJSON_CreateObject, 1);
+    will_return_always(__wrap_cJSON_AddStringToObject, 1);
+
+    // Adding data to JSON
+    expect_string(__wrap_cJSON_AddStringToObject, name, "old_status");
+    expect_string(__wrap_cJSON_AddStringToObject, string, "valid");
+    expect_string(__wrap_cJSON_AddStringToObject, name, "new_status");
+    expect_string(__wrap_cJSON_AddStringToObject, string, "obsolete");
+
+    // Printing JSON
+    will_return(__wrap_cJSON_PrintUnformatted, json_str);
+    expect_function_call(__wrap_cJSON_Delete);
+
+    // Calling Wazuh DB
+    expect_any(__wrap_wdbc_query_ex, *sock);
+    expect_string(__wrap_wdbc_query_ex, query, query_str);
+    expect_value(__wrap_wdbc_query_ex, len, WDBOUTPUT_SIZE);
+    will_return(__wrap_wdbc_query_ex, response);
+    will_return(__wrap_wdbc_query_ex, -100); // Returning any error
```

Comment:
It's true that -100 should be considered as an error, but, as these values are still not defined, and, in the future, -100 can be i.e. a warning code. I think it will be more standard to use a default code already implemented as -1. And it will not be required any comment to explain it.

Reply:
Fixed